core/domain/exp_domain.py

```diff
@@ -4268,6 +4318,12 @@ def _migrate_to_latest_yaml_version(
                 exploration_dict)
             exploration_schema_version = 28
 
+        if exploration_schema_version == 28:
+            exploration_dict['id'] = 'default'
```

Comment:
Why is `'default'` here?

Reply:
The Yaml of the exploration doesn't contain the exploration_id, but we need to have an exploration_id before the states function is called. That is why I had to assign some value to the exploration_dict['id'].
There's no specific reason for naming it default.

Reply:
That sounds wrong to me. I think we need to handle that case explicitly.

Reply:
I now have exp_id passed as an argument to the above function. And it works fine